rest/route/commit_queue.go

```diff
@@ -298,7 +298,7 @@ func (p *commitQueueAdditionalPatches) Parse(ctx context.Context, r *http.Reques
 func (p *commitQueueAdditionalPatches) Run(ctx context.Context) gimlet.Responder {
 	additional, err := data.GetAdditionalPatches(p.patchId)
 	if err != nil {
-		return gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "getting additional patches"))
```

Comment:
done

Reply:
https://github.com/evergreen-ci/evergreen/blob/7bdbaf53e0ef2eab15a818be0978859b02bd5276/rest/data/commit_queue.go#L369 Seems more like the line that should be modified.

Reply:
Alright, maybe I phrased this badly, so I'm gonna try again...

It would be good to keep the default as 500 instead of 400 unless we know it's for sure an issue due to user input. In cases like `GetAdditionalPatches` where the error could be due to user error (400) or Evergreen-side error (500), it's better to explicitly set the error response in the function itself and pass it back rather than rely on the default response code from `gimlet.NewErrorResponse` or `gimlet.NewInternalErrorResponse`. That means the returned values from all the returns in `data.GetAdditionalPatches` (like [this one](https://github.com/evergreen-ci/evergreen/blob/7bdbaf53e0ef2eab15a818be0978859b02bd5276/rest/data/commit_queue.go#L364) and [this one](https://github.com/evergreen-ci/evergreen/blob/7bdbaf53e0ef2eab15a818be0978859b02bd5276/rest/data/commit_queue.go#L384)) should explicitly return a `gimlet.ErrorResponse` with 400 rather than just a generic error type. Other cases in `GetAdditionalPatches` (like [this one](https://github.com/evergreen-ci/evergreen/blob/7bdbaf53e0ef2eab15a818be0978859b02bd5276/rest/data/commit_queue.go#L374)) are due to Evergreen internal DB state issues and not user error, so they should also explicitly set 500.

Reply:
@ablack12 I think that line is actually fine since that would only error for exceptional cases like being unable to reach the DB. That could be wrong though, since I don't know exactly which error return is being addressed in this PR.

Reply:
made some changes and it now returns 404 when things arent found

Reply:
Ah yes the line you grabbed is the one i was thinking of, did this too fast